src/text-buffer.js

```diff
@@ -1927,16 +1927,29 @@ class TextBuffer {
       try {
         await this.buffer.save(destination, this.getEncoding())
       } catch (error) {
-        const canEscalate = process.platform === 'darwin' || process.platform === 'linux'
-        if (error.code === 'EACCES' && destination === filePath && canEscalate) {
+        if (error.code !== 'EACCES' || destination !== filePath) throw error
+
+        const isMacOrLinux = process.platform === 'darwin' || process.platform === 'linux'
+        const isWindows = process.platform === 'win32'
+        if (isMacOrLinux) {
           const fsAdmin = require('fs-admin')
           try {
             await this.buffer.save(fsAdmin.createWriteStream(filePath), this.getEncoding())
           } catch (_) {
             throw error
           }
-        } else {
-          throw error
+        } else if (isWindows) {
+          const winattr = getPromisifiedWinattr()
+          const attrs = await winattr.get(filePath)
+          if (!attrs.hidden) throw error
+
+          try {
+            await winattr.set(filePath, { hidden: false })
+            await this.buffer.save(filePath, this.getEncoding())
+            await winattr.set(filePath, { hidden: true })
+          } catch (_) {
```

Comment:
What do you think about only checking for `windows` and put the rest of the code in an `else` block
Something like 
```js
if(isWindows) {
  const winattr = getPromisifiedWinattr()
  const attrs = await winattr.get(filePath)
  if (!attrs.hidden) throw error

  try {
    await winattr.set(filePath, { hidden: false })
    await this.buffer.save(filePath, this.getEncoding())
    await winattr.set(filePath, { hidden: true })
  } catch (_) {
    throw error
  }
} else {
  const fsAdmin = require('fs-admin')
  try {
    await this.buffer.save(fsAdmin.createWriteStream(filePath), this.getEncoding())
  } catch (_) {
    throw error
  }
}
```

Reply:
Makes total sense :) I'll add a fix later today